make docker fails in v8.1.0

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

### What version of TiKV are you using?
v8.1.0

### Steps to reproduce
Run `make docker`

### What did happened?
I get an error that rust-toolchain is missing.

```
docker build \
        -t pingcap/tikv:latest \
        -f Dockerfile \
        --build-arg GIT_HASH=ba73b0d92d94463d74543550d0efe61fa6a6f416 \
        --build-arg GIT_TAG=v8.1.0 \
        --build-arg GIT_BRANCH=heads/v8.1.0 \
        .
[+] Building 0.8s (15/35)
 => [internal] load build definition from Dockerfile                                                                                                                                                                                   0.0s
 => => transferring dockerfile: 4.04kB                                                                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                      0.0s
 => => transferring context: 566B                                                                                                                                                                                                      0.0s
 => [internal] load metadata for docker.io/pingcap/alpine-glibc:latest                                                                                                                                                                 0.7s
 => [internal] load metadata for docker.io/library/centos:7.6.1810                                                                                                                                                                     0.6s
 => CACHED [stage-2 1/4] FROM docker.io/pingcap/alpine-glibc@sha256:ac1e1b97fcba333a13c4adf1ec45de696a52260762a10dc2d6aadfbd81ae8e1d                                                                                                   0.0s
 => [prepare 1/5] FROM docker.io/library/centos:7.6.1810@sha256:62d9e1c2daa91166139b51577fe4f4f6b4cc41a3a2c7fc36bd895e2a17a3e4e6                                                                                                       0.0s
 => [internal] load build context                                                                                                                                                                                                      0.0s
 => => transferring context: 106.39kB                                                                                                                                                                                                  0.0s
 => CACHED [builder  2/22] RUN yum install -y epel-release &&     yum clean all &&     yum makecache                                                                                                                                   0.0s
 => CACHED [builder  3/22] RUN yum install -y centos-release-scl &&     yum install -y       devtoolset-8       perl cmake3 &&     yum clean all                                                                                       0.0s
 => CACHED [builder  4/22] RUN ln -s /usr/bin/cmake3 /usr/bin/cmake                                                                                                                                                                    0.0s
 => CACHED [builder  5/22] RUN curl -LO "https://github.com/protocolbuffers/protobuf/releases/download/v3.15.8/protoc-3.15.8-linux-x86_64.zip"                                                                                         0.0s
 => CACHED [builder  6/22] RUN unzip protoc-3.15.8-linux-x86_64.zip -d /usr/local/                                                                                                                                                     0.0s
 => CACHED [builder  7/22] RUN curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path --default-toolchain none -y                                                                                                                  0.0s
 => CACHED [builder  8/22] WORKDIR /tikv                                                                                                                                                                                               0.0s
 => ERROR [builder  9/22] COPY rust-toolchain ./                                                                                                                                                                                       0.0s
------
 > [builder  9/22] COPY rust-toolchain ./:
------
Dockerfile:64
--------------------
  62 |     # Install the Rust toolchain
  63 |     WORKDIR /tikv
  64 | >>> COPY rust-toolchain ./
  65 |     RUN rustup self update \
  66 |       && rustup set profile minimal \
--------------------
error: failed to solve: failed to compute cache key: failed to calculate checksum of ref bf81a72e-d9a6-401f-8f37-bd641a0e088d::jggocswmer3kqfj34x94mepvk: "/rust-toolchain": not found
make: *** [Makefile:406: docker] Error 1
```

`rust-toolchain` seems to have been changed to `rust-toolchain.toml` in this commit.
https://github.com/tikv/tikv/commit/26054366368e2288ce43468610b34d29e417c20f

Therefore, the following command will fail.
https://github.com/tikv/tikv/blob/55cafccd3ba359abd12aa88c03635e8866fb3cd9/Dockerfile#L64-L67

--- a/Dockerfile
+++ b/Dockerfile
@@ -58,13 +58,10 @@ ENV PATH /usr/local/bin/:$PATH
 # Install Rustup
 RUN curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path --default-toolchain none -y
 ENV PATH /root/.cargo/bin/:$PATH
+RUN rustup self update
 
-# Install the Rust toolchain
 WORKDIR /tikv
-COPY rust-toolchain ./
-RUN rustup self update \
-  && rustup set profile minimal \
-  && rustup default $(cat "rust-toolchain")
+COPY rust-toolchain.toml ./
 
 # For cargo
 COPY scripts ./scripts

--- a/Dockerfile.test
+++ b/Dockerfile.test
@@ -44,13 +44,10 @@ ENV PATH /usr/local/bin/:$PATH
 # Install Rustup
 RUN curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path --default-toolchain none -y
 ENV PATH /root/.cargo/bin/:$PATH
+RUN rustup self update
 
-# Install the Rust toolchain
 WORKDIR /tikv
-COPY rust-toolchain ./
-RUN rustup self update \
-  && rustup set profile minimal \
-  && rustup default $(cat "rust-toolchain")
+COPY rust-toolchain.toml ./
 
 RUN cargo install cargo-nextest --locked
 

--- a/rust-toolchain.toml
+++ b/rust-toolchain.toml
@@ -1,3 +1,4 @@
 [toolchain]
 channel = "nightly-2023-12-28"
 components = ["rustfmt", "clippy", "rust-src", "rust-analyzer"]
+profile = "minimal"

